app/queries/users_with_access.rb

```diff
@@ -0,0 +1,23 @@
+class UsersWithAccess
```

Comment:
I feel this would be neater as a scope on the `User` class. `queries` is a non-standard-Rails directory, other devs might not understand to look there.